addons/sale/views/sale_order_views.xml

```diff
@@ -479,7 +479,9 @@
                                         'no_open': True,
                                     }"
                                     domain="[('sale_ok', '=', True), '|', ('company_id', '=', False), ('company_id', '=', parent.company_id)]"
-                                    widget="sol_product_many2one"/>
+                                    widget="sol_product_many2one"
+                                    placeholder='Type to find a product... '
```

Comment:
```suggestion
                                    placeholder="Type to find a product..."
```
Unless there is a specific reason for the empty space in the end of the string, it can be removed ;).